### DISCUSSION TITLE
Docs development

### DISCUSSION THREAD

**Discussion by VividLemon** [cdmoro/bootstrap-vue-3#639] [Ideas]
@aceofwings 
How are you developing for the docs? I was going to try to work on them and create a basic template that contains the component references for each component. I think some people are confused if a component _exists_ because it is not in docs. I had one person say to me that they didn't want to use the project because they thought it was missing a component, when in reality, it existed the entire time. So, I was going to add in each component to the list, and only fill the component references. 

Now I am confused though about how I should begin this. It seems like there are three different ways that the references are created, one is raw like Spinners.md, another seems like it is pulled automatically from a resolveComponent, then it seems like others are inferred from .json files. 

It seems a bit messy, to be honest, I'm unsure how it's fetching data from the .json files, but when I add extra things like "slots" to the json file, it doesn't seem to do anything.

Going forward, how should the component references be made? I would like to standardize the component references and how they are made and at least get all the component references finished.

**Comment by aceofwings** [comment]
The Docs Reference Component is still incomplete as it cannot build the slots and emits sections. This can be solved by implementing those final features. 

Background 

Too save time and make easy changes to the reference, the docs reference component was created(Originating from bootstrap-vue).
Since our docs uses vuepress, it was easier to add in a plugin to grab the current component page name eg. Pagination.md and match up a json file with a certain object structure also named Pagination.json

Finally add ```<component-reference>``` to the page and you will have a partially rendered reference

Going Forward

Its hard to say because depending on how the component is written. The Vue SFC Script setup syntaxes does not compile quite concisely to DefineComponent causing issues with prop types. Again we also do not have the emit and slot functionality written out. Also I know we mentioned using actual TS props instead of primitive types( something I've spent some time looking into to automate).  I think it is easier and more maintainable to focus on perfecting the component-reference however it will need to be adjusted for some short comings I mentioned above.

**Reply by VividLemon** [reply]
Is there any way I can help? 

Would you prefer me to make a blank page for each component that contains the subsequent .json document that does contain the proper types etc?

**Reply by aceofwings** [reply]
Currently writing the slot implementation.  

You can look at Pagination.md and Pagination.json as examples. Most missing docs will not have a corresponding .json file. You will have to create both. I don't think it matters what component you start with. Just note for SFC script syntax components you will have to explicitly set the type.

**Reply by VividLemon** [reply]
In order to be consistent, should all files contain a json file with them? Since most components are setup syntax

**Comment by VividLemon** [comment]
@aceofwings Since an upcoming merger with BV is in the planning stages, it was discussed that the docs be transferred to a Nuxt3 app. Slightly to keep thinks consistent, slightly to support Nuxt3 (since pi0 and others work on Nuxt), and slightly because using Nuxt3 in docs can also provide a slight insight to the libraries compatibility with SSR. How do you feel about potentially rebuilding the docs to use Nuxt3 and Nuxt content? Not now, of course, but sometime in the future? 

I understand that vuepress also seems to use SSR as well, but I'm meaning to provide insight onto Nuxt3 compatibility, not necessarily vuepress

**Comment by aceofwings** [comment]
I have no problems with moving to nuxt3 for documentation support. I might not be fully part of the transition to nuxt3 depending the time involvement. Having docs built on a hierarchal markdown format makes things easier to add, remove and edit. If we were to go this route, then next move is  rebuild everything in a way similar to nuxt3 docs and depend on docusaurous as our new hierarchal system.  

Both Nuxt and Vuepress have SSR build features and similar plugin/hook features (In terms of create initialize and created hooks). Lets see this library and SSR evolves before moving over for the sake of nuxt3 & SSR support?

Ultimately it will be another project added to the long list of things.

Do we have a discussion or board to see the run down on what , who and how this merger is going to happen?

**Reply by VividLemon** [reply]
> depend on docusaurous as our new hierarchal system

I'm not exactly sure what that is, but I was expecting that Nuxt content could convert the md files in the same way, and possibly be hosted on Vercel, as I believe bootstrap-vue is, since they are a partner of bootstrap-vue. 

> Both Nuxt and Vuepress have SSR build features and similar plugin/hook features (In terms of create initialize and created hooks). Lets see this library and SSR evolves before moving over for the sake of nuxt3 & SSR support?

Anything works, I'm just mentioning it because you've put a great amount of work into the docs, most of everything can be ported over easily, but there would still need to be a bit of effort put in

> Do we have a discussion or board to see the run down on what , who and how this merger is going to happen?

No. There is a meeting planned at an unknown time, which will contain pi0, Carlos, and myself. It will likely be around noon, if you are in the US like me, as Carlos is in London, and pi0 in Amsterdam. I believe any collaborator should be fine to join, if you want to.

There is a small plan that I discussed with pi0 about what needs to happen: 
```
Plan:
- Move cdmoro/bootstrap-vue-3 to bootstrap-vue/bootstrap-vue-3
- Invite VividLemon and cdmoro to bootstrap-vue
- Check nuxt module issues (@pi0) 
- Try to migrate docs to Nuxt 3
- Check with illia to merge monorepos
- Track issues with pnpm
- Comparing Tests and List API breaking changes
```
There's more on the technical side of things, but one of the major things to work on is listing out the API breaking changes. The partyList.md documents some, but it isn't really a full migration guide, and I have been finding it hard to focus on building a full migration guide, especially when there is always upcoming changes to add before a full release.
